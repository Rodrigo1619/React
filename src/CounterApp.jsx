import {useState} from 'react'
import PropType from 'prop-types';

export const CounterApp = ({value})=>{
    //counter/setCounter no es nada mas como nosotros le quisimos poner
    const [counter, setCounter] = useState(value); //value es el valor inicial de nuestro hook y viene desde el main.jsx
    //lo que hace setCounter es cambiarnos nuestro valor de counter, debido a que es una constante y no se puede cambiar el valor
    const handleAdd = ()=>{
        setCounter(counter +1)
        //setCounter((c)=>c+1) //esta es otra forma de poder hacerlo 
    }

    //para restar
    const handleLess = () =>{
        setCounter(counter -1)
    }

    //Para restart
    const handleReset = ()=>{
        setCounter(value) //yo lo intente con counter = value pero esto da un error
    }

    return(
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>

            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleLess }>-1</button>
            <button onClick={ handleReset }>Reset</button>
        </>
    )
}

CounterApp.PropType ={
    value: PropType.number.isRequired
}