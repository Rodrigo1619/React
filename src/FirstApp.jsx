import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle, name}) => {
    //console.log(props);
    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            {/**< </h1>  */}
            <p>{subtitle}</p>
            <p>{name}</p>
        </>

    )
}
//Para poder agegarle tipado y que sea obligatorio mandar la propiedad
FirstApp.propTypes ={
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
}

FirstApp.defaultProps ={
    //title: 'Hola camaron sin cola',
    subtitle: 'subtitulo1',
    name: 'Rodrigo Molina'
}
