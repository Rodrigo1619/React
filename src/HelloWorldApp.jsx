export const HelloWorldApp = () => {

    return (
    <h1>camaron sin cola</h1>
    )
}
