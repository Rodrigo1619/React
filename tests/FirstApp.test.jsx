import { render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp/>', () => {
    test('Debe de hacer match con el snapshot', () => {
        const title ='Ola no camaron'
        const {container} =  render(<FirstApp title={title} /> );
        expect(container).toMatchSnapshot();
    })
    test('Debe de mostrar el titulo en h1', () => {
        const title ='Ola no camaron';
        const {container, getByText} =  render(<FirstApp title={title} /> );
        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain( title ); //mejor esto porque con el toBe cuenta hasta los espacios
    })
})