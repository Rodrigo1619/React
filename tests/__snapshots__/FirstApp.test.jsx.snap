// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Ola no camaron
  </h1>
  <p>
    123
  </p>
  <p>
    Rodrigo Molina
  </p>
</div>
`;
